File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/callback/BitmapCallback.java
package com.ashlikun.okhttputils.http.callback;

import android.graphics.Bitmap;
import android.widget.ImageView;

import com.ashlikun.okhttputils.http.convert.BitmapConvert;
import com.google.gson.Gson;

import okhttp3.Response;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/5/10 0010　下午 2:59
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */
public abstract class BitmapCallback extends AbsCallback<Bitmap> {

    private BitmapConvert convert;

    public BitmapCallback() {
        convert = new BitmapConvert();
    }

    public BitmapCallback(int maxWidth, int maxHeight) {
        convert = new BitmapConvert(maxWidth, maxHeight);
    }

    public BitmapCallback(int maxWidth, int maxHeight, Bitmap.Config decodeConfig, ImageView.ScaleType scaleType) {
        convert = new BitmapConvert(maxWidth, maxHeight, decodeConfig, scaleType);
    }

    @Override
    public Bitmap convertResponse(Response response, Gson gosn) throws Exception {
        Bitmap bitmap = convert.convertResponse(response, gosn);
        response.close();
        return bitmap;
    }
}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/callback/BitmapCallback.kt
package com.ashlikun.okhttputils.http.callback

import android.graphics.Bitmap
import android.widget.ImageView.ScaleType
import com.ashlikun.okhttputils.http.convert.BitmapConvert
import com.google.gson.Gson
import okhttp3.Response

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.16 19:39
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：
 */

abstract class BitmapCallback(
    val maxWidth: Int = 1000,
    val maxHeight: Int = 1000,
    val decodeConfig: Bitmap.Config = Bitmap.Config.ARGB_8888,
    val scaleType: ScaleType = ScaleType.CENTER_INSIDE
) : AbsCallback<Bitmap>() {
    private var convert: BitmapConvert =
        BitmapConvert(maxWidth, maxHeight, decodeConfig, scaleType)

    @Throws(Exception::class)
    override fun convertResponse(response: Response, gosn: Gson?): Bitmap {
        val bitmap = convert.convertResponse(response, gosn)
        response.close()
        return bitmap!!
    }
}